lib/airbrake-ruby/sync_sender.rb

```diff
@@ -33,18 +21,15 @@ def initialize(config)
     # @param [Airbrake::Notice] endpoint
     # @return [Hash{String=>String}] the parsed HTTP response
     def send(notice, endpoint = @config.endpoint)
-      response = nil
       req = build_post_request(endpoint, notice)
       https = build_https(endpoint)
 
-      begin
-        response = https.request(req)
-      rescue *HTTP_ERRORS => ex
-        @config.logger.error("#{LOG_LABEL} HTTP error: #{ex}")
-        return
-      end
-
+      response = https.request(req)
       Response.parse(response, @config.logger)
+
+    rescue => ex
```

Comment:
> The problem is, once the workers die, they are not resurrected. Are you open to changing that instead?

It sounds good to me in general, as a possible approach, but I reckon it's not needed right now (would be an overkill). As you said yourself "almost all of the exceptions will happen in https.request". I don't expect other code to fail, though. If it fails, then the library has a serious bug.

> the ultimate goal I have in mind isn't about keeping the workers alive, its about being able to make sure we only notify airbrake in a background thread

So if you have a broken worker (in a way as I described earlier), which you rescue on every `notify`, it would mean that you will never receive your error because it will be swallowed all the time. Currently, if workers die, you can still receive errors, which is IMO better than no errors at all.

As @vmihailenco pointed out earlier, Airbrake used to be synchronous by default for years. I understand your concerns but I believe it will be better to only rescue `https.request` (I don't expect any failures outside of that). If we will reveal that it's not enough, then we can always reconsider it. I am always open to feedback and I appreciate it a lot. So ping me any time for any [hopefully sensible, though] reason :P